### DISCUSSION TITLE
Apache Spark

### DISCUSSION THREAD

**Discussion by marijaselakovic** [crate/crate-clients-tools#193] [Ideas]
## About
Would be interesting to try out in a similar fashion: https://questdb.io/docs/third-party-tools/spark/

## Outcome
Tutorial on Apache Spark and CrateDB: [Getting started with Apache Spark and CrateDB: A step by step tutorial](https://community.cratedb.com/t/getting-started-with-apache-spark-and-cratedb-a-step-by-step-tutorial/1595)

## What's Inside
- [x] Install pyspark
- [x] Reading data from CrateDB
- [x] Writing data to CrateDB
- [x] If first tests succeed, try writing more complex types
- [x] Document in confluence
- [x] Write a tutorial
- [x] Publish tutorial on community

**Comment by amotl** [comment]
This has been converged into a community article.

https://community.cratedb.com/t/getting-started-with-apache-spark-and-cratedb-a-step-by-step-tutorial/1595

**Comment by amotl** [comment]
## Status
About Apache Spark specifically, we don't present much, most probably because connecting to Spark using vanilla pgJDBC works reasonably.

About advertisement matters, it's still very poor, so we should improve dearly. 🍀 

![Image](https://github.com/user-attachments/assets/c325a7bf-199e-43c9-b3d5-bad22b937a2b)

## Needs
I think the need for integrating optimally with Apache Spark is high, so we should elevate our attention.

/cc @surister, @zolbatar, @kneth, @ckurze

### 202x-

- https://github.com/crate/crate-clients-tools/issues/67
- https://github.com/crate/roadmap/issues/23
- https://github.com/crate/roadmap/issues/53
- https://github.com/crate/tech-content/issues/71
- https://github.com/crate/support/issues/482

### 2017-
- https://github.com/crate/crate/issues/5835
- https://github.com/crate/crate/issues/7118
- https://github.com/crate/crate/issues/8116
- https://github.com/crate/crate-jdbc/issues/228
- https://github.com/crate/support/issues/27
- https://github.com/crate/c-apporchid/issues/5
- https://github.com/crate/c-apporchid/issues/12
- https://github.com/crate/c-apporchid/issues/14
- https://github.com/crate/p-roosi/issues/3
- https://github.com/crate/p-roosi/issues/5

## Initiatives
I hope we will have a dedicated Spark track within upcoming initiatives.
- https://github.com/crate/roadmap/issues/91
- https://github.com/crate/roadmap/issues/81
- https://github.com/crate/writing/issues/24
